### PR TITLE
List of tabular models update

### LINKED ISSUE #1339: enh: Rework the list of tabular models and operations
The tabular models list should be reduced to important one:

Base candidates: CatBoost, LighGMB, XGBoost, Linear models. The promising ideas from LightAutoML/AutoGluon should be derived.

The low-priority models should be removed from default preset.

Related old issues:

- https://github.com/aimclub/FEDOT/issues/1217
- https://github.com/aimclub/FEDOT/issues/250
- https://github.com/aimclub/FEDOT/issues/312 - here is about ensembling methods
- https://github.com/aimclub/FEDOT/issues/1315 - here is about boostings

### PR DESCRIPTION
**Pull request by dmitryglhf**
This is a 🙋 feature or enhancement.

## Summary

- [ ] try to implement operations aggregations
- [ ] actualize the set of candidate models

## Context

  Closes #1339

### CODE REVIEW

**Comment by nicl-nno** [comment]
>try to implement operations aggregations

Думаю это лучше отдельным PR-ом

File: test/unit/optimizer/gp_operators/test_mutation.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def get_simple_linear_boosting_pipeline() -> Pipeline:
     node_pf = PipelineNode('poly_features', nodes_from=[node_scaling])
     node_rf = PipelineNode('rf', nodes_from=[node_pf])
     node_decompose = PipelineNode('class_decompose', nodes_from=[node_pf, node_rf])
-    node_linear = PipelineNode('linear', nodes_from=[node_decompose])
+    node_linear = PipelineNode('ridge', nodes_from=[node_decompose])
```
Comment (nicl-nno) [inline test/unit/optimizer/gp_operators/test_mutation.py:77]: А это ты просто так убрал или теперь нельзя вручную создать пайплайн с ridge?

**Review by nicl-nno — APPROVED** [review]

**Comment by dmitryglhf** [comment]
Results on benchmark with 1 hour 8c and 1 fold:

|    | Metric (mean)     |      Master |     PR#1380 |
|---:|:------------------|------------:|------------:|
|  0 | auc               |    0.901339 |    0.901033 |
|  1 | acc               |    0.86848  |    0.865966 |
|  2 | balacc            |    0.854555 |    0.852153 |
|  3 | logloss           |    0.339388 |    0.340543 |
|  4 | training_duration | 2101.31     | 1810.22     |

Метрика немного упала.

Сравнение первых 6ти датасетов:

| Dataset                              | Version | Pipeline                                                                 | Accuracy | AUC      | Balanced Accuracy | Log Loss  |  
|--------------------------------------|---------|--------------------------------------------------------------------------|----------|----------|-------------------|-----------|  
| Australian                           | Master  | logit, fast_ica, logit, pca, scaling, xgboost, lgbm, catboost, bernb     | 0.855072 | 0.941426 | 0.859508          | 0.449198  |  
|                                      | PR      | rf, scaling, poly_features, scaling, normalization                       | 0.869565 | 0.944822 | 0.872666          | 0.29653   |  
| Blood Transfusion Service Center     | Master  | mlp, normalization, poly_features, normalization, pca, fast_ica, normalization | 0.76     | 0.759747 | 0.595029          | 0.486645  |  
|                                      | PR      | logit, poly_features, scaling, resample                                 | 0.773333 | 0.774366 | 0.69883           | 0.59641   |  
| car                                  | Master  | rf                                                                      | 0.803468 |          | 0.865695          | 0.386452  |  
|                                      | PR      | catboost, logit                                                        | 0.774566 |          | 0.750098          | 0.40911   |  
| christine                            | Master  | catboost, scaling                                                      | 0.750923 | 0.83595  | 0.750923          | 0.505129  |  
|                                      | PR      | catboost, scaling                                                      | 0.750923 | 0.83595  | 0.750923          | 0.505129  |  
| cnae-9                               | Master  | logit, scaling                                                         | 0.962963 |          | 0.962963          | 0.141408  |  
|                                      | PR      | logit                                                                  | 0.962963 |          | 0.962963          | 0.138759  |  
| credit-g                             | Master  | logit, poly_features, scaling, isolation_forest_class                  | 0.82     | 0.850952 | 0.77619           | 0.449925  |  
|                                      | PR      | logit, poly_features, isolation_forest_class, scaling, poly_features   | 0.81     | 0.82619  | 0.769048          | 0.465911  |  

Судя по всему, на метрике сказался не обновленный список моделей, а то, что композер чаще выбирал операции с данными.

**Comment by nicl-nno** [comment]
А сохранилась история оптимизации для credit-g?

**Comment by nicl-nno** [comment]
И для car тоже интересно, конечно. Вроде rf же есть в начальных приближениях?

**Comment by dmitryglhf** [comment]
> А сохранилась история оптимизации для credit-g?

Не могу найти где посмотреть историю оптимизации в бенчмарке, вижу только обычные логи и мета-данные. В целом, я зафиксировал сид экспериментов, попробую повторить его локально.

> И для car тоже интересно, конечно. Вроде rf же есть в начальных приближениях?

Да интересно тут получилось, при этом и rf и catboost есть в начальных приближениях отдельно. Тоже посмотрю.

**Comment by nicl-nno** [comment]
>Не могу найти где посмотреть историю оптимизации в бенчмарке

Там вроде можно сохранять произвольные артефакты дополнительно к обязательным.

**Comment by dmitryglhf** [comment]
`car` history: 
[history.json](https://github.com/user-attachments/files/19531709/history.json)

`credit-g` history:
[history.json](https://github.com/user-attachments/files/19531711/history.json)

File: requirements.txt [hunk 2]
```diff
@@ -7,7 +7,7 @@ thegolem==0.4.1
 # Data
 numpy>=1.16.0, !=1.24.0
 pandas>=1.3.0; python_version >='3.8'
-dask-ml>=2024.4.4; python_version >= '3.8'
+dask-ml==2024.4.4; python_version >= '3.8'
```
Comment (DRMPN) [inline requirements.txt:10]: Я думаю, что можно попробовать закрепить версии для dask-ml в таком же формате.

```
sktime==0.16.1; python_version < '3.10'
sktime>=0.16.1; python_version >= '3.10'
```

Comment (DRMPN) [reply requirements.txt:10]: Тут вообще похоже проблема в dask-ml, который подтягивает dask, в котором есть следующая зависимость:

https://github.com/dask/dask/blob/4399aef0bfcba1a57d15cce458d6464fb7c0af2b/pyproject.toml#L35

Comment (dmitryglhf) [reply requirements.txt:10]: Да, я зафиксировал на fsspec==2023.1.0 и вроде завелось.

Comment (dmitryglhf) [reply requirements.txt:10]: Еще возможно в текущих requirements.txt зависимость dask-ml>=2023.5.0; python_version < '3.10' на самом деле не dask-ml, а просто dask, так как у него такая версия есть

**Review by Lopa10ko — APPROVED** [review]
